src/Xamarin.Android.Build.Tasks/Tasks/CilStrip.cs

```diff
@@ -55,7 +58,17 @@ public override bool Execute ()
 				string assemblyPath = Path.GetFullPath (assembly.ItemSpec);
 				string nonstripPath = Path.Combine (nonstripDir, Path.GetFileName (assemblyPath));
 
-				File.Move (assemblyPath, nonstripPath);
+				Log.LogDebugMessage ($"Moving {assemblyPath} to {nonstripPath}");
+
+				var srcmodifiedDate = File.GetLastWriteTimeUtc (assemblyPath);
+				var timestampFileDate = File.Exists (TimestampFile) ? File.GetLastWriteTimeUtc (TimestampFile) : DateTime.MinValue;
+
+				if (srcmodifiedDate < timestampFileDate) {
+					Log.LogDebugMessage ($"Skipping strip of IL for {assembly.ItemSpec}. Assembly has already been stripped.");
+					continue;
+				}
+
+				File.Copy (assemblyPath, nonstripPath, overwrite: true);
```

Comment:
I don't think so. `CopyIfChanged` uses a hash, in this case this task copies the `assemblyPath` to `nonstripPath` then the `cil-strip` exe copies the stripped `nonstripPath` file back over `assemblyPath`. So they will always have a different hash. 

The only way to check really is to check the timestamps against a stamp file of some sort.

Reply:
I don't think it does.

Reply:
The reason is RulCilStrip will write the new file back to `assemblyPath`, so I think we should be ok with `nonstripPath` having the last actual updated timestamp. It will make any time differences more apparent.